src/codegen/riscv64/macro-assembler-riscv64.cc

```diff
@@ -185,10 +184,13 @@ void MacroAssembler::RecordWriteField(Register object, int offset,
 
   Add64(dst, object, Operand(offset - kHeapObjectTag));
   if (emit_debug_code()) {
+    UseScratchRegisterScope temps(this);
+    Register scratch = temps.Acquire();
     BlockTrampolinePoolScope block_trampoline_pool(this);
```

Comment:
Actually, even with many uses of `BlockTrampolineScope`, it is possible that temp registers are ran out.

See [here](https://github.com/v8-riscv/v8/blob/93ccb7a80731191e3456cdad08ce14848594b65a/src/codegen/riscv64/macro-assembler-riscv64.cc#L2610)

I tried to remove `temps.hasAvailable() ? temps.Acquire() : t5;`, but without this code it would caused a failed check `DCHECK_NOT_NULL(available_);` which means here is no available temp registers.

I guessed it caused by too many nested functions.

The following `DCHECK(rs != scratch);` is used for this multiple used t5 register too.

Reply:
That will cause a problem though. If there are no temp registers available, then t5 is already in use, so just using it anyway is dangerous and likely to cause an error.

Reply:
Yes, that is.

So if we have more temp registers than 3(t3, t5, t6), it would be safe to use them.

Reply:
Thanks to @luyahan , the problem has been resolved. And I keep `BlockTrampolinePoolScope` just in case.